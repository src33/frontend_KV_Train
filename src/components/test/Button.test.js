import React from "react";
import { render, fireEvent } from "@testing-library/react";
import Button from "../Buttons";

describe("Check if Button works properly", () => {
   test("Check if button is rendered correctly", () => {
      const { getByTestId } = render(<Button />);
      const element = getByTestId("button-test-id");
      expect(element).toBeTruthy();
   });

   test("Check if text is displayed correctly", () => {
      const text = "Click";
      const { getByText } = render(<Button text={text} />);
      getByText(text);
   });

   test("check if onClick is triggered", () => {
      const onClick = jest.fn();

      const { getByTestId } = render(<Button handleClick={onClick} />);
      const element = getByTestId("button-test-id");

      fireEvent.click(element);

      expect(onClick).toHaveBeenCalled();
   });
   test("check if snapshots are matched properly", () => {
      const onClick = jest.fn();

      const text = "Click";

      const { asFragment } = render(<Button text={text} handleClick={onClick} />);

      expect(asFragment()).toMatchSnapshot();
   });
});
