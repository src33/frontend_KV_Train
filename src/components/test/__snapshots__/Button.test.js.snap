// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Check if Button works properly check if snapshots are matched properly 1`] = `
<DocumentFragment>
  <button
    data-testid="button-test-id"
  >
    Click
  </button>
</DocumentFragment>
`;
